sdk/core/Azure.Core/CHANGELOG.md

```diff
@@ -9,6 +9,7 @@
 ### Bugs Fixed
 
 ### Other Changes
+- `RequestFailedException` will not include the response content or headers in the message when the `IsError` property of the response is false.
```

Comment:
Yes, I think we should add it there also.

Reply:
Looking into this, we already don't log the content in `ClientResultException`

Reply:
Ah, fair.  I had thought we were logging the headers, but I just looked and we are not currently.

As a note we do have this outstanding request to add content logging: https://github.com/Azure/azure-sdk-for-net/issues/42823

It would be interesting to think about what we do there in light of this change, and if we were to implement inheritance of `RequestFailedException` from `ClientResultException`, could we easily implement the more secure logic in one place so we didn't get accidental drift there somehow.  Dunno how much of that you're interested in proactively solving as part of this change or leaving to future work -- we certainly don't need to block this PR on solutions to that problem, but if you spot good solutions easily, it might be nice to document them on that issue or elsewhere for whoever takes that item in later work.

Reply:
I think linking to this PR is good enough reference for now. I'd be happy to pick up the other work after this.